python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -872,6 +872,45 @@ def default_job_io_manager(init_context: "InitResourceContext"):
     return PickledObjectFilesystemIOManager(base_dir=instance.storage_directory())
 
 
+@io_manager(
+    description="Built-in filesystem IO manager that stores and retrieves values using pickling.",
+    config_schema={"base_dir": Field(StringSource, is_required=False)},
+)
+def default_job_io_manager_with_fs_io_manager_schema(init_context: "InitResourceContext"):
+    # support overriding the default io manager via environment variables
+    module_name = os.getenv("DAGSTER_DEFAULT_IO_MANAGER_MODULE")
+    attribute_name = os.getenv("DAGSTER_DEFAULT_IO_MANAGER_ATTRIBUTE")
+    silence_failures = os.getenv("DAGSTER_DEFAULT_IO_MANAGER_SILENCE_FAILURES")
+
+    if module_name and attribute_name:
+        from dagster._core.execution.build_resources import build_resources
+
+        try:
+            module = importlib.import_module(module_name)
+            attr = getattr(module, attribute_name)
+            check.invariant(
+                isinstance(attr, IOManagerDefinition),
+                "DAGSTER_DEFAULT_IO_MANAGER_MODULE and DAGSTER_DEFAULT_IO_MANAGER_ATTRIBUTE must specify an IOManagerDefinition",
+            )
+            with build_resources({"io_manager": attr}) as resources:
+                return resources.io_manager
+        except Exception as e:
+            if not silence_failures:
+                raise
+            else:
+                warnings.warn(
+                    f"Failed to load io manager override with module: {module_name} attribute: {attribute_name}: {e}\n"
+                    "Falling back to default io manager."
+                )
+    from dagster._core.storage.fs_io_manager import PickledObjectFilesystemIOManager
+
+    # normally, default to the fs_io_manager
+    base_dir = init_context.resource_config.get(
+        "base_dir", init_context.instance.storage_directory()
```

Comment:
Some of the places where we were directly setting fs_io_manager have tests fail if we don't include this block. So I think it's a breaking change if we don't keep them separate, with this different schema and different base directory